crates/core/src/protocols/fri/prove.rs

```diff
@@ -382,7 +380,10 @@ where
 			.fold_arities()
 			.get(self.round_committed.len() + 1)
 			.map(|log| 1 << log)
-			.unwrap_or_else(|| self.params.rs_code().inv_rate());
+			.unwrap_or_else(|| {
+				1 << log2_strict_usize(folded_codeword.len())
+					.saturating_sub(self.params.rs_code().log_inv_rate())
+			});
```

Comment:
```suggestion
			.unwrap_or_else(|| {
				1 << self.params.n_final_challenges()
			});
```
indeed, this too can be simplified; and you can get rid of `log2_strict_usize